src/scaler/manager.rs

```diff
@@ -441,16 +406,43 @@ where
                                     // NOTE(thomastaylor312): We could find that this strategy actually
                                     // doesn't tear down everything or leaves something hanging. If that is
                                     // the case, a new part of the reaper logic should handle it
-                                }
-                                Notifications::BackoffScaler(name) => {
-                                    if let Some(scalers) = self.get_scalers(&name).await {
-                                        trace!(%name, "Handling backoff scaler event");
-                                        futures::future::join_all(
-                                            scalers
-                                                .iter()
-                                                .map(|scaler| scaler.backoff(self.backoff_sender.clone())),
-                                        )
-                                        .await;
+                                },
+                                Notifications::RegisterExpectedEvents{ name, scaler_id, triggering_event } => {
+                                    trace!(%name, "Computing and registering expected events for manifest");
+                                    if let Some(scaler) = self.get_specific_scaler(&name, &scaler_id).await {
+                                        if let Some(event) = triggering_event {
+                                            let parsed_event: Event = match event.try_into() {
+                                                Ok(e) => e,
+                                                Err(e) => {
+                                                    error!(error = %e, %name, "Unable to parse given event");
+                                                    continue;
+                                                }
+                                            };
+                                            if let Err(e) = scaler.handle_event(&parsed_event).await {
+                                                error!(error = %e, %name, %scaler_id, "Unable to register expected events for scaler");
+                                            }
+                                        } else if let Err(e) = scaler.reconcile().await {
```

Comment:
Not sure I follow this. Why do we call `reconcile` if there's no triggering event? (And why do we not call reconcile when removing an expected event?)

Reply:
This is cheating and I'll add a comment to clarify. Basically, if this is a backoff wrapped scaler, it will cause the expected events to be added to the list. If it isn't we just return the commands and do nothing with them